cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -594,14 +610,93 @@ impl Lockfile {
     /// re-fetched.
     #[instrument(skip_all, fields(progress = "Merging environment includes"))]
     fn merge_manifest(
+        flox: &Flox,
         manifest: &Manifest,
-        _seed_lockfile: Option<&Lockfile>,
-    ) -> Result<(Manifest, Option<Compose>), EnvironmentError> {
+        seed_lockfile: Option<&Lockfile>,
+        include_fetcher: &IncludeFetcher,
+        merger: ManifestMerger,
+    ) -> Result<(Manifest, Option<Compose>), RecoverableMergeError> {
         if manifest.include.environments.is_empty() {
             return Ok((manifest.clone(), None));
         }
 
-        todo!()
+        if !flox.features.compose {
+            return Err(RecoverableMergeError::Catchall(
+                indoc! {"Cannot handle [include] when compose feature flag is disabled.
+                Use 'flox config --set-bool features.build true' to enable composition."}
+                .to_string(),
+            ));
+        }
+
+        debug!("Composing included environments");
+
+        // Fetch included manifests we don't already have in seed_lockfile.
+        // Note that we have to preserve the order of the includes in the
+        // manifest.
+        let mut locked_includes: Vec<LockedInclude> = vec![];
+        for include_environment in &manifest.include.environments {
```

Comment:
Suggestion: it would be nice to log which environment you're fetching so that when it fails you know where you were in this loop.

Reply:
Done